### PR TITLE
Update protobuf to 3.7.1

### PR DESCRIPTION
**Pull request by sjdrc**
I just merged your patches with upstream. Compiles fine and I'm able to use protoc to compile proto files within cmake.

### CODE REVIEW

File: python/setup.py [hunk 1]
```diff
@@ -202,6 +202,18 @@ def get_option_from_sys_argv(option_str):
             r'macosx-[0-9]+\.[0-9]+-(.+)', r'macosx-10.9-\1',
             util.get_platform())
 
+    # https://developer.apple.com/documentation/xcode_release_notes/xcode_10_release_notes
```
Comment (ruslo) [inline python/setup.py:205]: Not related to hunterization, please remove, see:
- https://docs.hunter.sh/en/latest/creating-new/patch.html

**Review by ruslo — CHANGES_REQUESTED** [review]
Please rebase your changes to get rid of the merge commits.

File: python/setup.py (hunk 1 above)
Comment (sjdrc) [reply python/setup.py:205]: I didn't add this, it's from upstream.

**Comment by sjdrc** [comment]
Can you please give me some guidance on exactly what you want me to do here? I have made no changes to the repository, all I want to do it apply the 18 changes you made on top of 3.6.1.3 to the top of 3.7.1 upstream. I have attempted to rebase using the commandline but I am unfamiliar with the workflow and do not think I have done it correctly.

File: python/setup.py (hunk 1 above)
Comment (ruslo) [reply python/setup.py:205]: There is no such change in 3.7.1 tag. At this moment `hunter-3.7.1` branch is a clean 3.7.1 upstream version.

**Comment by ruslo** [comment]
> all I want to do it apply the 18 changes you made on top of 3.6.1.3

Why 18?

See `hunter-3.6.1.3` branch:
- https://github.com/hunter-packages/protobuf/commits/hunter-3.6.1.3

Just cherry-pick and test next changes:
- https://github.com/hunter-packages/protobuf/commit/dfa809c7a021afe8a8642486bd55dfaf9dd64d9c
- https://github.com/hunter-packages/protobuf/commit/f1fc9b3f13393f5c33f1ad0758bce5dbb20483a5
- https://github.com/hunter-packages/protobuf/commit/0809903c78916983b9147f6b617949d8074341f4
- https://github.com/hunter-packages/protobuf/commit/ad15aac3debe117c03fe84991d07501a8cb122cd
- https://github.com/hunter-packages/protobuf/commit/90eacfab75b00165792522ac4d2c16db4517ec68

**Comment by sjdrc** [comment]
I'll reopen this.

**Comment by ruslo** [comment]
> I'll reopen this

You can do a force push; there is no need to open a new pull request.

**Pull request closed without merging**